### DISCUSSION TITLE
using ItemsService to update items not logging activity

### DISCUSSION THREAD

**Discussion by ranaharoon3222** [directus/directus#11466] [Archive]
I am using hooks to update items using "ItemsService"   but when I use  "ItemsService.updateOne"  it's updating everything correctly but didn't post to activity so I can't see any activity log.  But when I use Axios it works fine. Also, how can I authenticate in hooks to get Bearer Token so I can include it in Axios headers?

```js
   const orderService = new ItemsService('orders', {
       schema: await getSchema(),
   });

   await orderService.updateOne(key, {
       total: order.data.total - price,
       totalWithOutShipping: order.data.totalWithOutShipping - price,
   });
```

<details>
<summary>More Info</summary>

### What version of Directus are you using?

9.5.1

### What version of Node.js are you using?

v14.16.1

### What database are you using?

mysql

### What browser are you using?

chrome

### What operating system are you using?

windows 10

### How are you deploying Directus?

running locally

</details>

**Comment by azrikahar** [comment]
Moved this to discussion as it's more of a extensions/hooks usage or question rather than an inherent bug in Directus 👍
 
> I am using hooks to update items using "ItemsService" but when I use "ItemsService.updateOne" it's updating everything correctly but didn't post to activity so I can't see any activity log. But when I use Axios it works fine.

It's because in your `ItemsService('orders', ...)`, the accountability was not passed. If we check out ItemsService under the hood, we can see this block of code, particularly the comment on it saying activity will only be logged if there's an accountability + the collection has activity tracking enabled.

https://github.com/directus/directus/blob/b8c4317cb1cd9a8ed0eeb1ad4db35f8771588624/api/src/services/items.ts#L434-L451

You can refer the docs and see how to use `accountability` that is available in the context object: https://docs.directus.io/extensions/hooks/

>  Also, how can I authenticate in hooks to get Bearer Token so I can include it in Axios headers?

Would you mind elaborating what endpoint(s) is the Axios going to call with said token? As you're inside hooks or extensions in general, you should be able to perform various business logics via all the services available: https://github.com/directus/directus/tree/main/api/src/services, rather than using Axios. However if there's a use case not achievable with the services listed above, feel free to let us know.

**Reply by ranaharoon3222** [reply]
thanks a lot!!!

But Directus should include documentation for item services in docs. so devs can know how they can use it

**Reply by azrikahar** [reply]
Totally! We are tracking this over here: https://github.com/directus/directus/issues/10523
